Add template for Jakarta EE 9 for further discussion

kwsutter opened the pull request:
Signed-off-by: Kevin Sutter <kwsutter@gmail.com>

Starting with a simple update for the Jakarta EE 9 _index.md record.  Let's play with this content and format before applying it across the board.

Besides the information for the Release Plan, I also moved the Compatible Implementation section up to push the boring Ballots to the bottom.  The Ballots take up too much room.  And, if we add Progress Reviews to this section, or if we have to do a re-do, this Ballot section will just continue to grow and make these pages unusable.  I'm hoping that putting it at the bottom of the page is sufficient.

bshannon commented:
This looks pretty good.

There's still the question of what to put in the ballot section in those cases where no actual
voting will be done, only "approval  by consensus".  It seems that we want to record that
somehow.

kwsutter commented:
> There's still the question of what to put in the ballot section in those cases where no actual
> voting will be done, only "approval by consensus". It seems that we want to record that
> somehow.
Bill, I have updated the Plan Review section with "comments".  Comments in MD are kind of ugly, but this syntax seems to work.  The default statement will point the reader at the Jakarta EE 9 Plan Review ballot.  For the standalone Plan Reviews, the comments recommend a cut-and-paste from the Release Review section.  Would this suffice?

waynebeaton commented:
If you set the date in the header to a future date, then the page will not be rendered by Hugo until after that date. IMHO, this is a handy and consistent way to have a place to accumulate information about the release (like the results of the just-completed plan review) in one place without adding potentially confusing entries on the specifications page, or the need to copy information from another location later.

I'm not sure how Hugo will deal with 2020-mm-dd. I recommend that we use the actual projected release date and update it as necessary.

waynebeaton commented:
Kevin, can you add these ballot results into the platform page?

Representative | Representative for: | Vote
-- | -- | --
Kenji Kazumura, Michael DeNicola | Fujitsu | 1
Dan Bandera, Kevin Sutter, Alasdair Nottingham | IBM | 1
Bill Shannon, Ed Bratt, Dmitry Kornilov | Oracle | 1
Steve Millidge, Mark Wareham | Payara | 1
Scott Stark, Mark Little | Red Hat | 1
David Blevins, Richard Monson-Haefel | Tomitribe |  
Ivar Grimstad | EE4J PMC | 1
Alex Theedom | Participant Members | 1
Werner Keil | Committer Members | 1
  | Total | 8

kwsutter commented:
> Kevin, can you add these ballot results into the platform page?

Will do, but I think the Total number of votes is now 9...  :-)

kwsutter commented:
> If you set the date in the header to a future date, then the page will not be rendered by Hugo until after that date. IMHO, this is a handy and consistent way to have a place to accumulate information about the release (like the results of the just-completed plan review) in one place without adding potentially confusing entries on the specifications page, or the need to copy information from another location later.

I forgot about that feature of Hugo.  I kind of like that.  If I set the date as 2020-06-30 for all of these templates, then it will render for our pre-deployment viewing.  But, it won't be public until each Specification Versions gets approved.  That keeps the jakarta.ee site clean.
> 
> I'm not sure how Hugo will deal with 2020-mm-dd. I recommend that we use the actual projected release date and update it as necessary.

Hugo is not processing 2020-mm-dd as a "future date" since the original pages are already rendering and made public...  Here's Persistence 3.0 as an example...
https://jakarta.ee/specifications/persistence/3.0/

kwsutter commented:
Here the two updated pages thus far.  If these look good, then I can proliferate to the other components.  Thanks for the reviewing!
https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/platform/9/ 
https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/annotations/2.0/ 

P.S.  I also noticed that Hugo automatically adjusts the ordering of the Specification Versions based on the ```date:``` value in the header.  That's nice!

But, now that brings up a valid point on the value to be used in this field.  I was originally thinking the proposed release date (2020-06-30).  But, that would prevent these pages from rendering on the public site until then...  And, they do contain valuable data for the Plan reviews.  So, my new suggestion is to use 2020-01-15 since that's when the Plan Review ballot completed.

kwsutter commented:
Requesting another look-see.  I've adjusted the Annotations _index.md to use some comment blocks.  This allows me to hide some material that doesn't apply yet, but still gives the general expected format for the future Release Review PRs.  It makes the .md a little busy, but the output looks clean.  Let me know what you think.  You can compare the "full" Platform template with the "partial" Annotation template.  Thanks!
https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/platform/9/
https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/annotations/2.0/

kwsutter commented:
Shoot!  The rendering engine of Hugo doesn't match the rendering engine in my IDE...  Scrap this idea for now.  Let me experiment some more and post when I'm ready for a real review.  Sorry!

kwsutter commented:
@bshannon @waynebeaton @dblevins @smillidge 
I'm skipping the idea of using block comments in the .md files...  Can't get anything to work with Hugo except for the individual comment lines, so I reverted my changes to just using those.  If somebody knows of something that would work for block comments, let me know.  But, I've tried everything...  As all of my previous commits can attest to...  :-)

I think what I have now will work for all of the individual components.  I'll propagate the changes once I get a plus-one or two... Thanks!

https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/platform/9/
https://deploy-preview-132--jakartaee-specifications.netlify.com/specifications/annotations/2.0/

@waynebeaton I corrected the list of reps you provided in your [earlier comment](https://github.com/jakartaee/specifications/pull/132#issuecomment-574908332).  I only listed the primary and alternate participants.  And, I replaced Richard with Cesar for Tomitribe.  This is consistent with [our Jakarta EE About page](https://jakarta.ee/about/#specifications-committee).

kwsutter commented:
Since I'm not hearing any additional comments, I'll go with lazy consensus and start making the necessary updates across the board.  We can always tweak this again down the road.  Thanks for the reviews!

kwsutter commented:
Well, that was more of a pain than I expected...  :-). Could only do so much automatically.  Oh well, not that bad.  I'll double check everything after the pre-deployment build finishes and then I'll commit this material to be ready for Jakarta EE 9 activity.

kwsutter commented:
The pre-deployment view looked pretty good.  I'm going to go ahead with the merge and we'll let each Project adjust as needed.  They are close enough.

FYI, I left the date in the _index.md file as 2020-MM-DD for the three Projects which are expected to have a separate Release Review -- Activation, Enterprise Beans, and Restful Web Services.  Although it sounds like Restful Web Services might just go with a simple release for Jakarta EE 9.  I'll double check before merging.  Thanks!
